homeassistant/components/deluge/config_flow.py

```diff
@@ -0,0 +1,101 @@
+"""Config flow for Deluge Client."""
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_NAME,
+    CONF_PASSWORD,
+    CONF_PORT,
+    CONF_SCAN_INTERVAL,
+    CONF_USERNAME,
+)
+from homeassistant.core import callback
+
+from . import get_api
+from .const import DEFAULT_NAME, DEFAULT_PORT, DEFAULT_SCAN_INTERVAL, DOMAIN
+from .errors import CannotConnect, PasswordError, UserNameError, UnknownError
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_NAME, default=DEFAULT_NAME): str,
+        vol.Required(CONF_HOST): str,
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+        vol.Required(CONF_PORT, default=DEFAULT_PORT): int,
+    }
+)
+
+
+class DelugeFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a Deluge config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry):
+        """Get the options flow for this handler."""
+        return DelugeOptionsFlowHandler(config_entry)
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user."""
+        errors = {}
+
+        if user_input is not None:
+
+            for entry in self.hass.config_entries.async_entries(DOMAIN):
+                if entry.data[CONF_HOST] == user_input[CONF_HOST]:
+                    return self.async_abort(reason="already_configured")
+                if entry.data[CONF_NAME] == user_input[CONF_NAME]:
+                    errors[CONF_NAME] = "name_exists"
+                    break
+
+            try:
+                await get_api(self.hass, user_input)
+
+            except UserNameError:
+                errors[CONF_USERNAME] = "user_error"
+            except PasswordError:
+                errors[CONF_PASSWORD] = "password_error"
+            except (CannotConnect, UnknownError):
+                errors["base"] = "cannot_connect"
+
+            if not errors:
+                return self.async_create_entry(
+                    title=user_input[CONF_NAME], data=user_input
+                )
+
+        return self.async_show_form(
+            step_id="user", data_schema=DATA_SCHEMA, errors=errors
+        )
+
+    async def async_step_import(self, import_config):
+        """Import from Deluge client config."""
+        import_config[CONF_SCAN_INTERVAL] = import_config[CONF_SCAN_INTERVAL].seconds
+        return await self.async_step_user(user_input=import_config)
+
+
+class DelugeOptionsFlowHandler(config_entries.OptionsFlow):
```

Comment:
Let's drop the options flow from the PR. We don't want to make this configurable and it keeps the PR easier.

Reply:
I would like to say that the update interval is necessary because if we set the default 120 the user might want a faster update rate.
This will only be an option that the user can ignore if they don't understand it.